_posts/2017/2017-02-20-continuous-documentation-with-ansible.md

```diff
@@ -0,0 +1,220 @@
+---
+layout: post
+title: "Continuous Documentation with Ansible"
+date: "2017-02-20 06:39:20"
+image: blog-header/scrum-board.jpg
+categories: tech-stories agile devops docker jenkins ansible
+authors: ["Thomas H."]
+---
+
+## Introduction
+In recent years there has been a [steady trend towards cloud based deployments](http://www.softwareadvice.com/buyerview/deployment-preference-report-2014/) of applications and services.  
+This increases the distributed nature of the systems we develop, which in turn raises the complexity and effort
+required to develop and manage (i.e. provision resources, build, test, configure, deploy and monitor) them.  
+This is especially true, as during this transition phase the environment is heterogeneous and different methods of building and deploying software are combined (e.g. RPM packages and Docker containers).
+
+At the same time, delivery cycles are expected to shorten because of the [opportunity cost of not releasing software](http://robertbaillie.blogspot.de/2014/10/the-opportunity-cost-of-delaying.html)
+and the [Agile principles](http://agilemanifesto.org/principles.html) of customer collaboration and early feedback.
+
+This post will outline how [Ansible](https://www.ansible.com/) can help in approaching these challenges.
+It also includes a short discussion on the relation of Ansible to other tools supporting continuous delivery and
+scalability of the development process, [Docker](https://www.docker.com/) and [Jenkins](https://jenkins.io/).
+
+## Documentation vs. Automation
+<img src="/assets/images/blog/blog-dilbert-agile.gif" alt="Dilbert comic agile" style="width: 100%;"/>
+
+The way we think about documentation [is changing](https://devops.com/documenting-devops-agile-automation-and-continuous-documentation/).  
+For the documentation of application software it is generally accepted that it should be generated from code, to leave as little possibility for deviation as possible.  
+In the case of "management code" this is more difficult, because the things being modelled are often heterogeneous and not as focussed in terms of their implementation as for example a REST-API or a library / module.  
+The concept of [using declarative formats for automation](https://12factor.net/#introduction) can fill this gap.
+
+## Structure of Ansible projects
+As there are many sources of Ansible best practices to be found on the web, instead of writing another one, I would like to assume the viewpoint of someone reading Ansible code, rather than writing it.
+Note that I am using the term "Ansible project" without implying a certain structure in terms of source code repositories.  
+
+### Playbooks
+[Playbooks](http://docs.ansible.com/ansible/playbooks.html) are the entry point when looking at an Ansible project.  
+A common convention is two categorize playbooks into two kinds.
```

Comment:
noticed that there is a small typo, it should be: A common convention is **to** categorize playbooks into two kinds.